remote/src/main/java/com/spotify/metrics/remote/OkRemote.java

```diff
@@ -89,4 +91,17 @@ public void onResponse(Call call, Response response) throws IOException {
         });
         return result;
     }
+
+    @Override
+    public void waitForAllCalls() {
+        Dispatcher dispatcher = client.dispatcher();
```

Comment:
@parmus That was my first thought but I think this will prevent the queued calls from being executed. They are put in a different queue first if the execution limit was reached and later on handed to the executor. We need all to be executed.